smart-contracts/contracts/PublicLock.sol

```diff
@@ -387,6 +395,33 @@ contract PublicLock is ILockCore, ERC165, IERC721, IERC721Receiver, Ownable {
     refundPenaltyDenominator = _refundPenaltyDenominator;
   }
 
+  /**
+  * @dev Used to disable lock before migrating keys and/or destroying contract
+   */
+  function disableLock()
+    external
+    onlyOwner
+    onlyIfAlive
+  {
+    emit Disable(address(this));
+    isAlive = false;
+  }
+
+  /**
+  * @dev Used to clean up old lock contracts from the blockchain
+  * TODO: add a check to ensure all keys are INVALID!
+   */
+  function destroyLock()
+    external
+    onlyOwner
+  {
+    require(isAlive == false, "Not allowed to delete an active lock");
+    emit Destroy(address(this), this.balance, msg.sender);
+    selfdestruct(msg.sender);
+    // Note we don't clean up the `locks` data in Unlock.sol as it should not be necessary
+    // and leaves some data behind which may be helpful.
```

Comment:
Can you specify which data is left behind here?

Reply:
Done.